swarm/storage/mru/resource.go

```diff
@@ -245,45 +238,55 @@ func (h *Handler) SetStore(store *storage.NetStore) {
 // It implements the storage.ChunkValidator interface
 func (h *Handler) Validate(addr storage.Address, data []byte) bool {
 
+	if data[0] == 0 && data[1] == 0 && len(data) > common.AddressLength {
+		//metadata chunk
+		rootAddr, _ := metadataHash(data)
+		valid := bytes.Equal(addr, rootAddr)
+		if !valid {
+			log.Warn("Invalid root metadata chunk")
+		}
+		return valid
+	}
+
 	// does the chunk data make sense?
 	// This is not 100% safe evaluation, since content addressed data can coincidentally produce data with length header matching content size
-	signature, period, version, name, parseddata, _, err := h.parseUpdate(data)
+	mru, err := parseUpdate(data)
 	if err != nil {
 		log.Warn("Invalid resource chunk")
 		return false
 	}
 
 	// Check if signature is valid against the chunk address
 	// Since we force signatures to be used, we can know be sure that this is valid data
-	digest := h.keyDataHash(addr, parseddata)
-	publicKey, err := crypto.SigToPub(digest.Bytes(), signature[:])
+	ownerAddr, err := mru.Verify(addr)
```

Comment:
`SignedResourceUpdate.Verify()` deals with making sure the `SignedResourceUpdate` has a valid signature